File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hm=new HashMap<>();
        
        for(int i=0;i<nums.length;i++){
            int sum=target-nums[i];

            if(hm.containsKey(sum)){
                return new int[] {i,hm.get(sum)};
            }
            hm.put(nums[i],i);
        }

        return new int[] {0,0};
    }
}

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hm = {}
        for i in range(len(nums)):
            sum_needed = target - nums[i]
            if sum_needed in hm:
                return [i, hm[sum_needed]]
            hm[nums[i]] = i
        return [0, 0]